sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorPersister.java

```diff
@@ -126,95 +130,198 @@ public CuratorPersister build() {
      * @param serviceSpec the service for which data will be stored
      */
     public static Builder newBuilder(ServiceSpec serviceSpec) {
-        return newBuilder(serviceSpec.getZookeeperConnection());
+        return newBuilder(serviceSpec.getName(), serviceSpec.getZookeeperConnection());
     }
 
     /**
      * Creates a new {@link Builder} instance which has been initialized with reasonable default values.
      */
-    public static Builder newBuilder(String zkConnectionString) {
-        return new Builder(zkConnectionString);
+    @VisibleForTesting
+    public static Builder newBuilder(String serviceName, String zookeeperConnection) {
+        return new Builder(serviceName, zookeeperConnection);
     }
 
     @VisibleForTesting
-    CuratorPersister(CuratorFramework client) {
+    CuratorPersister(String serviceName, CuratorFramework client) {
+        this.serviceRootPath = CuratorUtils.getServiceRootPath(serviceName);
         this.client = client;
         this.client.start();
     }
 
     @Override
```

Comment:
Reviewer note: this file has a bit more churn than strictly necessary as its functions were reordered to match the order of the `Persister` interface. The effective changes are:
- Addition of custom root-delete handling in `deleteAll()` (ported from `DefaultStateStore`)
- Catching `Exception`/`KeeperExceptions`s and converting them into the appropriate `PersisterException`s
- Prefixing received paths with `/dcos-service-foo` before accessing Curator itself